### PR TITLE
Update default sphinx version

### LINKED ISSUE #4208: autoclass not working, but no warnings or errors in log
## Details

* Read the Docs project URL: http://epifx.readthedocs.io/en/latest/getting_started.html#forecast-summaries
* Build URL (if applicable): https://readthedocs.org/projects/epifx/builds/7306664/
* Read the Docs username (if applicable): robmoss

## Expected Result

When I build the documentation locally, I see:

epifx-docs-local

## Actual Result

When it builds on RTD, the result is:

epifx-docs-rtd

The only error or warning in the build log is:

`copying static files... WARNING: html_static_path entry '/home/docs/checkouts/readthedocs.org/readthedocs/templates/sphinx/_static' does not exist`

So I have no idea what might be causing this.

### PR DESCRIPTION
**Pull request by stsewd**
It contains a lot of bug fixes http://www.sphinx-doc.org/en/master/changes.html#release-1-7-5-released-may-29-2018

Closes #4208

### CODE REVIEW

**Comment by ericholscher** [comment]
I wonder if we should just do `<1.7`?

**Comment by humitos** [comment]
I'd like to have `<1.8` but "you never know". Anyway, maybe instead of increasing the version when it breaks for some reason, we could decrease (or pin) when it breaks.

:+1: on using `<1.8`

File: readthedocs/rtd_tests/tests/test_doc_building.py [hunk 1]
```diff
@@ -1152,7 +1152,7 @@ def test_install_core_requirements_sphinx(self):
         requirements_sphinx = [
             'commonmark==0.5.4',
             'recommonmark==0.4.0',
-            'sphinx==1.7.5',
+            'sphinx<1.8',
```
Comment (ericholscher) [inline readthedocs/rtd_tests/tests/test_doc_building.py]: Feels like we shouldn't be checking for this in a test, as it makes it really brittle. Should we remove this test?

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_doc_building.py]: Looking at it more, it looks like we should be checking for what we actually care about (eg. `'sphinx' in called_arguments` not "check it called the exact versions from our code"

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_doc_building.py]: That makes sense, should I change all the tests here?

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_doc_building.py]: Sure. I think it's a small enough change to be fine.

**Comment by ericholscher** [comment]
👍

**Pull request merged**